src/module/ovs/ovs.rs

```diff
@@ -45,6 +46,33 @@ impl Collector for OvsCollector {
         cmd.register_module::<OvsCollectorArgs>(ModuleId::Ovs)
     }
 
+    // Check if the OvS collector can run. Some potential errors are silenced,
+    // to avoid returning an error if we can't inspect a given area for some
+    // reasons.
+    fn can_run(&self) -> Result<()> {
+        let inspector = inspect::inspector()?;
+
+        // First check the OvS kernel module is available.
+        if let Ok(kconf) = inspector.kernel.get_config_option("CONFIG_OPENVSWITCH") {
+            match kconf {
+                None => bail!("Incompatible kernel configuration: CONFIG_OPENVSWITCH=n"),
+                Some("m") => {
+                    if inspector.kernel.is_module_loaded("openvswitch") == Some(false) {
+                        bail!("Kernel module 'openvswitch' is not loaded")
+                    }
+                }
+                _ => (),
+            }
+        }
+
+        // Then check if the openvswitch user process is running.
+        if !proc::find_process_by_name("ovs-vswitchd")? {
```

Comment:
Technically, the module can run even if ovs-vswitchd is not present. Only `--track` will not work. Not sure if we need to express that somehow here.

Also, ovs kernel module is ns-aware. So it's common to run several! ovs-vswitchd instances in containers, all of them using the same kernel module. Again, --track will not work but still kernel module can be traced

Reply:
> > Technically, the module can run even if ovs-vswitchd is not present. Only `--track` will not work. Not sure if we need to express that somehow here.
> 
> So maybe we should have the cli as an argument to `can_run` because some capabilities might depends on some flags. And then we can add the check only if `--track` was used.
> 

Having the cli as an argument might be cool (although we might end have with a chicken-and-egg situation).

> Out of curiosity, can the ovs module handle packets if the userspace part is not running?
> 

Yes. Temporarily when you restart the userspace part, the kernel works headless. Upcalls are dropped but currently established traffic flows

> > Also, ovs kernel module is ns-aware. So it's common to run several! ovs-vswitchd instances in containers, all of them using the same kernel module. Again, --track will not work but still kernel module can be traced
> 
> Interesting, so if we want to do this correctly we should also check if those processes match the current ns. (Isn't something we should do in `Process` too to avoid the `PID_ALL` case?).

The common use case (at least for me) is to use [kind](https://kind.sigs.k8s.io/). In such cases the module will just not find the process by cmd and --track will not work. Still it's good to see what goes through the kernel datapath. Even more when we have ns filters. 
To find the right process we'd need to jump into the pid-ns of the container. Not sure how to do it cleanly so I don't think we need to make "--track" work on kind today. For now, it's just another case in which the userspace might not be available but kernel-only ovs module should be able to work.

Reply:
Cool. BTW, I think it's OK to fail if "--track" has been provided and "ovs-vswitchd" is not present (or more than one is) but I guess we'll fail when the module is initialized anyway and the error is not very bad.